vault/logical_system.go

```diff
@@ -2905,6 +2914,9 @@ func (b *SystemBackend) pathInternalUIMountRead(ctx context.Context, req *logica
 		Data: mountInfo(me),
 	}
 	resp.Data["path"] = me.Path
+	if ns.ID != me.Namespace().ID {
```

Comment:
Wouldn't `MatchingMountEntry` always return the mounts for the namespace from the passed in context? In what cases would this check apply?

Reply:
This fixes being able to reach into sub-namespaces and access the mount info. So if you wanted to do:
```
vault kv put ns/kv/foo bar=baz
```